floats/floats_test.go

```diff
@@ -919,8 +1038,57 @@ func TestNearestWithinSpan(t *testing.T) {
 			value:  7.249,
 			idx:    2,
 		},
+		{
+			length: 4,
+			lower:  math.Inf(-1),
+			upper:  math.Inf(1),
+			value:  math.Copysign(0, -1),
+			idx:    0,
+		},
+		{
+			length: 5,
+			lower:  math.Inf(-1),
+			upper:  math.Inf(1),
+			value:  0,
+			idx:    2,
+		},
+		{
+			length: 4,
+			lower:  math.Inf(-1),
+			upper:  math.Inf(1),
+			value:  0,
+			idx:    2,
+		},
+		{
+			length: 4,
+			lower:  math.Inf(-1),
+			upper:  math.Inf(1),
+			value:  math.Inf(1),
+			idx:    2,
```

Comment:
Why is the expected `idx` not 3? It seems wrong that `0` and `inf` lead to same `idx`.

Reply:
The (hypothetical) slice is []float64{-inf, -inf, inf, inf}. -0 is closer to -inf, +0 is closer to inf. This way they both return idx = 2.

Reply:
Thanks, seeing the hypothetical slice explains it. But documenting the behavior regarding all the possible combinations of corner values and for the user to understand it, should this function really be in Gonum?

Reply:
Why the hypothetical slice is `[]float64{-inf, -inf, inf, inf}`? If `n` is odd, does it have 0 in the middle? Why?

Reply:
It seems like the only logical behavior since it's halfway between the two.

`Span` shouldn't be used if there are NaN and Inf involved. It is a very reasonable function without those limiting cases. However, the input cannot be finite numbers, so we have to have behavior with inf and nan inputs. We can either panic with those inputs, or try to do something sensible. It's not clear to me which is worse here, but neither is good.

Reply:
> logical behavior since it's halfway between the two.

That's probably the essence of my unease with this. I don't think it's logical because I don't think there's anything halfway between -inf and +inf. Or that it is clear how a hypothetical slice from -inf to 3 (for example) looks like (I really don't know, I guess it would be something like `[]float{-inf, -inf, -inf, 3}`).

The output seems to be useful only if the inputs are finite numbers, so the user will not be able to blindly pass numbers in and use the result anyway, so I would vote for panicking. But much work has went into this and there are rules with reasoning behind them, so if they are documented, then LGTM too.

Reply:
SGTM

Reply:
I will remove the documentation of the special cases.